File: app/src/main/java/com/ead/project/dreamer/data/models/EmbedServer.java
package com.ead.project.dreamer.data.models;

import javax.inject.Inject;

public class EmbedServer extends Server {

    @Inject
    public EmbedServer(String embeddedUrl,Player player) {
        super(embeddedUrl,player);

        if (isDownloading) return;

        this.player = player;
        this.isDirect = false;

        this.url = setupEmbeddedUrl(embeddedUrl);

        if (checkIfVideoIsAvailable()) {
            addDefaultVideo();
        }
    }

    protected String setupEmbeddedUrl(String embeddedUrl) {
        return embeddedUrl;
    }

    protected Boolean checkIfVideoIsAvailable() {
        return false;
    }
}

File: app/src/main/java/com/ead/project/dreamer/data/models/EmbedServer.kt
package com.ead.project.dreamer.data.models

import android.content.Context

open class EmbedServer(context: Context,url : String): Server(context, url) {
    override val isDirect: Boolean
        get() = false

    override suspend fun onExtract() {
        if (isAvailable()) {
            addDefault()
        }
    }

    protected open fun isAvailable(): Boolean {
        return false
    }
}
